utils/common/src/main/kotlin/EnvironmentVariableFilter.kt

```diff
@@ -0,0 +1,117 @@
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.common
+
+/**
+ * An object providing functionality to filter environments that are passed to newly created processes.
+ *
+ * For many tasks, ORT spawns new processes using the [ProcessCapture] class. When creating a new process, the child
+ * process per defaults inherits all environment variables from the parent. This could impose a security risk, for
+ * instance if logic in build scripts could theoretically access sensitive information stored in environment
+ * variables, such as database or service credentials.
+ *
+ * To reduce this risk, this object filters the environment variables passed to child processes based on the
+ * following criteria:
+ * - Patterns on variable names can be defined to determine variables with sensitive information. The object provides
+ *   some default patterns to match variables containing strings like "PASS", "USER", "TOKEN", etc.
+ * - There is an allow list to include variables even if they are matched by a pattern.
+ *
+ * So in order to determine whether a specific variable "E" can be passed to a child process, this filter applies the
+ * following steps:
+ * - If E is contained in the allow list, it is included.
+ * - Otherwise, E is included if and only if it is not matched by one of the exclusion patterns.
+ */
+object EnvironmentVariableFilter {
+    /**
+     * A set with patterns for variable names that are denied by default. All variables containing one of these
+     * strings (ignoring case) are not propagated to child processes.
+     */
+    val DEFAULT_DENY_PATTERNS = setOf(
+        "key",
+        "pass",
+        "pwd",
+        "token",
+        "user"
+    )
+
+    /** A set with known variable names that are accepted despite they are matched by deny patterns. */
+    val DEFAULT_ALLOW_NAMES = setOf(
+        "CARGO_HTTP_USER_AGENT",
+        "COMPOSER_ALLOW_SUPERUSER",
+        "CONAN_LOGIN_ENCRYPTION_KEY",
+        "CONAN_LOGIN_USERNAME",
+        "CONAN_PASSWORD",
+        "CONAN_USERNAME",
+        "CONAN_USER_HOME",
+        "CONAN_USER_HOME_SHORT",
+        "DOTNET_CLI_CONTEXT_ANSI_PASS_THRU",
+        "GIT_ASKPASS",
+        "GIT_HTTP_USER_AGENT",
+        "GRADLE_USER_HOME",
+        "HACKAGE_USERNAME",
+        "HACKAGE_PASSWORD",
+        "HACKAGE_KEY",
+        "PWD",
+        "USER",
+        "USERPROFILE"
+    )
+
+    /** Stores the current deny patterns used by this filter. */
+    private var denyPatterns = DEFAULT_DENY_PATTERNS.toPatterns()
```

Comment:
BTW, this line confirms me that "patters" is the wrong word for the variable, because if they already were patters, there would be no point in converting them `toPatterns()`.